### PR TITLE
Unify Trigger and Clear Action button

### PR DESCRIPTION
**Pull request by pierrejeambrun**
The goal is:
- Unify the action button behavior for clear / trigger and any further actions.
- Support a `withText=true` mode for a full button
- Support a `withText=false` mode for a small, compact action style button with tooltip

Screenshot 2024-12-19 at 23 33 24
Screenshot 2024-12-19 at 23 33 33
Screenshot 2024-12-19 at 23 33 49
Screenshot 2024-12-19 at 23 34 09

### CODE REVIEW

File: airflow/ui/src/components/ui/ActionButton.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  const ButtonComponent: FC<ButtonProps> = withText ? Button : IconButton;
+
+  return (
+    <Tooltip content={actionName} disabled={Boolean(withText)}>
+      <ButtonComponent
+        aria-label={actionName}
+        colorPalette={withText ? colorPalette : "blue"}
+        onClick={onClick}
+        size={withText ? "md" : "sm"}
+        variant={withText ? variant : "ghost"}
```
Comment (pierrejeambrun) [inline airflow/ui/src/components/ui/ActionButton.tsx:52]: `variant` and `colorPalette` (or any other custom styling for 'withText') I don't like but couldn't find an elegant way.

Meaning that it's not super easy to use an `<ActionButton />` and customize style for `withText` and `compact` (withText=false) version distinctly.

Comment (pierrejeambrun) [reply airflow/ui/src/components/ui/ActionButton.tsx:52]: The final `TriggerDAGButton` and `ClearRunButton` props are clean and easy to use though. Making sure those actions have consistant display over different places, it's just that intermediate layer that I don't really like.

**Review by jscheffl — APPROVED** [review]
Cool! Thanks for the consolidation! Will make the next 20 buttons much better!

File: airflow/ui/src/components/ui/ActionButton.tsx (hunk 1 above)
Comment (jscheffl) [reply airflow/ui/src/components/ui/ActionButton.tsx:52]: I am not a pro in React... is some kind of inheritance also possible (compared to use an abstract component in another component as middle layer)?

Comment (pierrejeambrun) [reply airflow/ui/src/components/ui/ActionButton.tsx:52]: That's not really critical. Maybe someone will find a more flexible way. We don't really need it for now.

Moving forward.

Comment (pierrejeambrun) [reply airflow/ui/src/components/ui/ActionButton.tsx:52]: If things didin't change there's no inheritance in react but Composition. I tried thinking about that but I didn't find an ideal approach.

**Pull request merged**